packages/flutter/test/material/scaffold_test.dart

```diff
@@ -1676,6 +1676,90 @@ void main() {
     await tester.pumpWidget(buildFrame(false, null));
     expect(tester.getSize(find.byKey(bodyKey)), const Size(800.0, 500.0));
   });
+
+  group('Throws FlutterError', () {
+    testWidgets(
+        'Call to Scaffold.of() without context', (WidgetTester tester) async {
+      await tester.pumpWidget(MaterialApp(
+          home: Builder(
+              builder: (BuildContext context) {
+                Scaffold.of(context).showBottomSheet<void>((
+                    BuildContext context) {
+                  return Container();
+                });
+                return Container();
+              }
+          )
+      ));
+      final dynamic exception = tester.takeException();
+      expect(exception, isFlutterError);
+      final FlutterError error = exception;
+      expect(error.toStringDeep(),
+          'FlutterError\n'
+              '   Scaffold.of() called with a context that does not contain a\n'
+              '   Scaffold.\n'
+              '   No Scaffold ancestor could be found starting from the context\n'
+              '   that was passed to Scaffold.of(). This usually happens when the\n'
+              '   context provided is from the same StatefulWidget as that whose\n'
+              '   build function actually creates the Scaffold widget being sought.\n'
+              '   There are several ways to avoid this problem. The simplest is to\n'
+              '   use a Builder to get a context that is "under" the Scaffold. For\n'
+              '   an example of this, please see the documentation for\n'
+              '   Scaffold.of():\n'
+              '     https://api.flutter.dev/flutter/material/Scaffold/of.html\n'
+              '   A more efficient solution is to split your build function into\n'
```

Comment:
These are probably hints. you can verify that they are hints by calling error.toDiagnosticNode
and getting the last couple children of the diagnostics node. The last two children should be hints with exactly this text.
That way you have a test that will pass with the new version of the code but fail with the old one.